### ISSUE TITLE
Use Synchronized Output sequences

### ISSUE THREAD

**Issue by valpackett** [helix-editor/helix#731]
#### Describe your feature request

To avoid any possible visual glitches like flickering, many modern terminal emulators (WezTerm, Kitty, iTerm2, foot, mintty, …) support DEC mode `2026` as described in:

https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036

to allow applications like text editors to mark complete frames with begin and end markers.

Helix should ensure these are used.

**Comment by kirawi** [comment]
It looks like `termwiz` supports it, but there's no available release yet for the commits that implemented it.

**Comment by the-mikedavis** [comment]
If it isn't already implemented in crossterm (https://github.com/crossterm-rs/crossterm) it might make a good contribution there. Currently we use crossterm for terminat manipulation. I think in the long run we want to switch to termwiz though which already supports it

**Comment by kirawi** [comment]
There was #1847, but it looks like it's stalled.

**Comment by the-mikedavis** [comment]
This was added to crossterm recently in https://github.com/crossterm-rs/crossterm/pull/756 but we will need detection (similar to #4939). I'll post a patch upstream to add the detection function

**Comment by k4lizen** [comment]
Any progress on this? When I move and resize my windows around in kitty I get flicker even worse than the one in this issue: https://github.com/kovidgoyal/kitty/discussions/4395

**Comment by k4lizen** [comment]
https://github.com/user-attachments/assets/948f135a-64ea-40e4-8bc8-270f219d5419

**Comment by kirawi** [comment]
Someone still needs to add detection upstream.

**Comment by k4lizen** [comment]
Could someone who understands more about this than me open a feature request for the detection upstream so the people working on crossterm at least know it's keeping this issue open

**Comment by k4lizen** [comment]
@the-mikedavis I can still reproduce my flicker video from above, should I open a new issue?

**Comment by the-mikedavis** [comment]
Yeah, making window resizing smoother should be tracked separately. Synchronized output sequences are a necessary step but we will probably need smarter handling in Helix as well. During resizes we get a lot of window resize events and we should debounce these and only re-render when we haven't received one for a while. I believe what's happening there is that we're re-rendering every so often and by the time the render finishes the window has already shrunk and a new render is needed.
